manager/integration/tests/test_scheduling.py

```diff
@@ -899,8 +895,18 @@ def test_data_locality_basic(client, core_api, volume_name, pod, settings_reset)
 
     volume4 = volume4.updateDataLocality(dataLocality="disabled")
     volume4 = volume4.updateReplicaCount(replicaCount=2)
-    wait_for_volume_healthy(client, volume4_name)
-    volume4 = wait_for_volume_replica_count(client, volume4_name, 2)
+
+    running_replica_count = 0
+    for _ in range(RETRY_COUNTS):
+        volume4 = client.by_id_volume(volume4_name)
+        running_replica_count = 0
+        for r in volume4.replicas:
+            if r.failedAt == "" and r.running is True:
+                running_replica_count += 1
+        if running_replica_count == 2:
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert running_replica_count == 2
```

Comment:
Must explicitly wait for the volume to have 2 running replicas because if we don't, the volume might be in the state with 1 failed replica and 1 running replica.